clang/test/Driver/sycl-specific-args-diagnostics.cpp

```diff
@@ -0,0 +1,151 @@
+
+// Warning should be emitted when using -fsycl-instrument-device-code without -fsycl
+// RUN:   %clang -### -fsycl-instrument-device-code  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-INS-DEV -DOPT=-fsycl-instrument-device-code %s
+// RUN:   %clang_cl -### -fsycl-instrument-device-code  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-INS-DEV -DOPT=-fsycl-instrument-device-code %s
+// WARNING-INS-DEV: warning: argument unused during compilation: '[[OPT]]' [-Wunused-command-line-argument]
+
+// Warning should be emitted when using -fsycl-libspirv-path without -fsycl
+// RUN:   %clang -### -fsycl-libspirv-path=libspirv.bc  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-LIBSPRIV  %s
+// RUN:   %clang_cl -### -fsycl-libspirv-path=libspirv.bc  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-LIBSPRIV %s
+// WARNING-LIBSPRIV: warning: argument unused during compilation: '-fsycl-libspirv-path=libspirv.bc' [-Wunused-command-line-argument]
+
+// Warning should be emitted when using -fsycl-max-parallel-link-jobs without -fyscl
+// RUN:   %clang -### -fsycl-max-parallel-link-jobs=4  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-PARALLEL %s
+// RUN:   %clang_cl -### -fsycl-max-parallel-link-jobs=4  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-PARALLEL %s
+// WARNING-PARALLEL: warning: argument unused during compilation: '-fsycl-max-parallel-link-jobs=4' [-Wunused-command-line-argument]
+
+// Warning should be emitted when using -fsycl-optimize-non-user-code without -fsycl
+// RUN:   %clang -### -fsycl-optimize-non-user-code  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-OPTIMIZE -DOPT=-fsycl-optimize-non-user-code %s
+// RUN:   %clang_cl -### -fsycl-optimize-non-user-code  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-OPTIMIZE -DOPT=-fsycl-optimize-non-user-code %s
+// WARNING-OPTIMIZE: warning: argument unused during compilation: '[[OPT]]' [-Wunused-command-line-argument]
+
+// Warning should be emitted when using -fsycl-unnamed-lambda without -fsycl
+// RUN:   %clang -### -fsycl-unnamed-lambda  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-UNNAMED-LAMBDA -DOPT=-fsycl-unnamed-lambda %s
+// RUN:   %clang_cl -### -fsycl-unnamed-lambda  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-UNNAMED-LAMBDA -DOPT=-fsycl-unnamed-lambda %s
+// WARNING-UNNAMED-LAMBDA: warning: argument unused during compilation: '[[OPT]]' [-Wunused-command-line-argument]
+
+// Warning should be emitted when using -fsycl-use-bitcode without -fsycl
+// RUN:   %clang -### -fsycl-use-bitcode  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-USE-BITCODE -DOPT=-fsycl-use-bitcode %s
+// RUN:   %clang_cl -### -fsycl-use-bitcode  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-USE-BITCODE -DOPT=-fsycl-use-bitcode %s
+// WARNING-USE-BITCODE: warning: argument unused during compilation: '[[OPT]]' [-Wunused-command-line-argument]
```

Comment:
To further improve - you can use a generic `WARNING-UNUSED` check-prefix and apply the `{{OPT]]` behavior to all of the unused command line checks you have.